lib/foreman/gettext/support.rb

```diff
@@ -42,19 +42,31 @@ def self.add_text_domain(locale_domain, locale_dir)
           :report_warning => false
       end
 
-      def self.register_human_localenames
-        original_locale = FastGettext.locale
-        FastGettext.class.class_eval { attr_accessor :human_available_locales }
-        FastGettext.human_available_locales = []
-        FastGettext.default_available_locales.sort.each do |locale|
-          FastGettext.locale = locale
-          # TRANSLATORS: Provide locale name in native language (e.g. English, Deutsch or Portugues)
-          human_locale = _("locale_name")
-          human_locale = locale if human_locale == "locale_name"
-          FastGettext.human_available_locales << [human_locale, locale]
+      # Loading all available locales to get the language translation is slow
+      # (2 seconds for 20 languages), so let's lazy load that.
+      FastGettext.class_eval do
+        attr_accessor :mutex, :locales
+
+        self.mutex = Mutex.new
```

Comment:
This would mean the mutex is always allocated, even if it is never used.  Can there be a case where we never try to get a list of all the locales?

Reply:
What you mean by "allocated"? It must be created during class loading for the code to be safe.

Reply:
If you used
```ruby
def mutex
  @mutex ||= Mutex.new
end
```

then it would be created on first use while still ensuring the code is safe since it is a class instance variable

Reply:
I am not sure creating a mutex is atomic, so this could lead to two copies of mutex. Neither my code is safe, however probability of hitting two concurrent threads creating the same mutex is low when it's executed during startup. That's my thinking.